src/reactionsystem.jl

```diff
@@ -756,6 +756,55 @@ function ReactionSystem(iv; kwargs...)
     ReactionSystem(Reaction[], iv, [], []; kwargs...)
 end
 
+
+"""
+    remake_ReactionSystem_internal(rs::ReactionSystem; 
+        default_reaction_metadata::Vector{Pair{Symbol, T}} = Vector{Pair{Symbol, Any}}()) where {T}
+
+Takes a `ReactionSystem` and remakes it, returning a modified `ReactionSystem`. Modifications depend
+on which additional arguments are provided. The input `ReactionSystem` is not mutated.
+
+Arguments:
+- `rs::ReactionSystem`: The `ReactionSystem` which you wish to remake.
+- `default_reaction_metadata::Vector{Pair{Symbol, T}}`: A vector with default `Reaction` metadata values.
+    Each metadata in each `Reaction` of the updated `ReactionSystem` will have the value desiganted in
+    `default_reaction_metadata` (however, `Reaction`s that already have that metadata designated will not
+    have their value updated).
+"""
+function remake_ReactionSystem_internal(rs::ReactionSystem;  default_reaction_metadata = [])
+    # Updates the metadata for all reactions (equation are ignored).
+    updated_equations = [set_default_metadata(rx, default_reaction_metadata) for rx in reactions(rs)]
+    updated_reactions = [set_default_metadata(rx, default_reaction_metadata) for rx in reactions(rs)]
+    @set! rs.eqs = updated_equations
+    @set! rs.rxs = updated_reactions
```

Comment:
This is both flattening implicitly by using `reactions(rs)` instead of `get_rxs(rs)`, and also is dropping any non-reaction equations. Remember, `equations(sys)` and `reactions(rs)` include all subsystems' equations and reactions too. To only pull from the top-level system, use `get_eqs(sys)` and `get_rxs(sys)`. Those are the field accessors.

It is also creating multiple copies of the reactions in the new `eqs` and `rxs` fields instead of having the vectors point to the same `Reaction`s in memory. I think you want something more like
```suggestion
    eqtransform(eq) = eq isa Reaction ? set_default_metadata(eq, default_reaction_metadata) : eq
    updated_equations = map(eqtransform, get_eqs(rs))
    @set! rs.eqs = updated_equations
    @set! rs.rxs = Reaction[rx for rx in updated_equations if rx isa Reaction]
```